server/catgenome/src/main/java/com/epam/catgenome/manager/reference/ReferenceManager.java

```diff
@@ -145,24 +156,30 @@
     public Reference registerGenome(final ReferenceRegistrationRequest request) throws IOException {
 
         final String name;
+        String path = request.getPath();
         if (request.getType() == null) {
             request.setType(BiologicalDataItemResourceType.FILE);
         }
         if (request.getType() == BiologicalDataItemResourceType.GA4GH) {
             name = request.getName();
         } else {
-            name = parse(request.getPath(), request.getName());
+            name = parse(path, request.getName());
         }
         // prepares to start processing of a reference genome: generates ID, creates a directory
         // to store data for a genome
         final Long referenceId = referenceGenomeManager.createReferenceId();
         final Reference reference = new Reference(referenceId, name);
-        reference.setPath(request.getPath());
+        reference.setPath(path);
+        reference.setSource(path);
         reference.setPrettyName(request.getPrettyName());
         if (!request.isNoGCContent()) {
             fileManager.makeReferenceDir(reference);
         }
         reference.setType(request.getType());
+        if (GenbankUtils.isGenbank(path)) {
+            path = genbankToFasta(reference);
```

Comment:
since we can have resorceType == S3 or Azure, but here we rewrite whole reference to local file, I believe we should set here type == FILE here